backend/src/acidwatch_api/models/example_adapter.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+from typing import Annotated, Any, Self, override
+from acidwatch_api.models.base import BaseAdapter, Setting, Concs, Metadata, Settings
+from pydantic import BaseModel
+
+
+class ExampleSettings(BaseModel):
+    spontaneously_combust: Annotated[int, Setting(
+        # Label that will be shown to the user
+        label="Spontaneously combust",
+        # Description
+        description="The rate at which atoms will spontaneously disappear",
+        # Minimum (optional)
+        min=0,
+        # Maximum (optional)
+        max=100,
+        # Which unit to show this as
+        unit="%",
+        # Default value
+        default=50,
+    )]
+
+    spontaneously_combust_two: Annotated[int, Setting(
+        # Label that will be shown to the user
+        label="Spontaneously combust",
+        # Description
+        description="The rate at which atoms will spontaneously disappear",
+        # Which unit to show this as
+        unit="%",
+    )]
+
+
+class ExampleAdapter(BaseAdapter):
+    """This is an example on how to write model adapters"""
+
+    class Settings:
+        sponta: Annotated[int, Setting]
+
+        temperature: int = Param(label="Temperature", min=200, max=400, unit="%")
+
+    model_id = "example"
+
+    # Every model requires a human-readable model name. This text will be
+    # displayed in the frontend.
+    model_name = "Example"
+
+    # Every model has a set of initial concentrations of substances.
+    #
+    # This function must return a dict that maps a valid chemical substance
+    # (eg. 'HNO2', but not 'ASDF') to a floating point value between 0 and
+    # 1,000,000 (in ppm) or None, indicating it's possible for the user to
+    # supply it, but it's disabled by default.
+    #
+    # The exception is CO2, which must not be specified as it's the solvent.
+    concs = {
+        "HNO": 10,
+    }
```

Comment:
I kinda dislike that each model needs their own default value.. Does it not make more sense for the models to simply determine:

`accepted_components = ["HNO", "H2O"]`

And AcidWatch sets a default value for the most familiar ones, and 0 for the rest? Reduces complexity a bit for this, and might make it simpler for the user when they switch between models (as in, all models have same defaults).